src/scmrepo/git/backend/pygit2.py

```diff
@@ -459,7 +496,63 @@ def fetch_refspecs(
         progress: Callable[["GitProgressEvent"], None] = None,
         **kwargs,
     ) -> Mapping[str, SyncStatus]:
-        raise NotImplementedError
+        from pygit2 import GitError
+
+        if isinstance(refspecs, str):
+            refspecs = [refspecs]
+
+        try:
+            remote: "Remote" = self.repo.remotes[url]
+        except ValueError:
+            remote_name = "fetch"
+            self.repo.remotes.set_url(remote_name, url)
+            remote = self.repo.remotes[remote_name]
+        except KeyError:
+            raise SCMError(f"'{url}' is not a valid Git remote or URL")
+
+        if os.name == "nt" and remote.url.startswith("file://"):
```

Comment:
Not sure if should I put this code here or in the tests? Looks like `pygit2` doesn't support this format on the Windows platform.

Reply:
it should support it https://github.com/libgit2/libgit2/blob/1327dbcf2a4273a8ba6fd978db5f0882530af94d/src/libgit2/transport.c#L35

Reply:
> it should support it https://github.com/libgit2/libgit2/blob/1327dbcf2a4273a8ba6fd978db5f0882530af94d/src/libgit2/transport.c#L35

But the test result is that on Mac or Linux it supports it, but not on Windows. I guess it's a bug of pygit2.

Reply:
It might be that libgit2 or pygit2 expect windows slashes instead of posix ones after the `file://` protocol, or it might be that the `file://` prefix is supposed to be stripped as you are doing here. But in regular CLI git there is actually different behavior depending on whether you provide a `file://` URL or a raw local path (without `file://`), so we should double check what the problem is.

Please try to take a look into what pygit2 is doing in this situation

Reply:
Hadn't found any issues related to it. Let me create a issue for them https://github.com/libgit2/pygit2/issues/1187

Reply:
This could still be due to not creating a proper `Remote()` object for the raw URL. As we discussed, we should be using [`git_remote_create_anonymous`](https://libgit2.org/libgit2/#HEAD/group/remote/git_remote_create_anonymous) to generate our remote here.

```py
remote_name = "fetch"
self.repo.remotes.set_url(remote_name, url)
remote = self.repo.remotes[remote_name]
```

This is likely going to cause undefined behavior because we are trying to call `set_url` on a remote that doesn't actually exist (unless the user has already defined their own remote named `fetch`, in which case we are incorrectly modifying an existing git remote instead of creating our own temporary one)

Reply:
Looks like `[git_remote_create_anonymous](https://libgit2.org/libgit2/#HEAD/group/remote/git_remote_create_anonymous)` are not implemented in `pygit2`, I just make a temporary solution.

Reply:
Right, as we discussed though we should really consider adding it to `pygit`